ios/Classes/SwiftFlutterWeatherSensorPlugin.swift

```diff
@@ -1,14 +1,76 @@
 import Flutter
 import UIKit
+import Combine
 
-public class SwiftFlutterWeatherSensorPlugin: NSObject, FlutterPlugin {
+@available(iOS 13.0, *)
+public class SwiftFlutterWeatherSensorPlugin: NSObject, FlutterPlugin, FlutterStreamHandler {
+  private var eventSink: FlutterEventSink? = nil
+  private var sinkSubscription: AnyCancellable?
+    
   public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "flutter_weather_sensor_plugin", binaryMessenger: registrar.messenger())
+    let methodChannel = FlutterMethodChannel(name: "flutter_weather_sensor_plugin", binaryMessenger: registrar.messenger())
+    let eventChannel = FlutterEventChannel(name: "flutter_weather_sensor_plugin/readings", binaryMessenger: registrar.messenger())
+      
     let instance = SwiftFlutterWeatherSensorPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addMethodCallDelegate(instance, channel: methodChannel)
+    eventChannel.setStreamHandler(instance)
   }
 
   public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    result("iOS " + UIDevice.current.systemVersion)
+    if (call.method == "startSensorReadings") {
+      startSensorReadings()
+      result("success")
+    } else if (call.method == "stopSensorReadings") {
+      stopSensorReadings()
+      result("success")
+    } else if (call.method == "set") {
+      if let args = call.arguments as? Dictionary<String, Any>,
+        let interval = args["readingInterval"] as? NSInteger {
+          setInterval(interval)
+        } else {
+          result(FlutterError.init(code:"invalidArgs", message: "Reading interval must be an integer.", details: "Expected 1 Int arg."))
+        }
+      result("success")
+    } else if (call.method == "collectReadings") {
+      collectReadings()
+      result("success")
+    } else {
+      result(FlutterMethodNotImplemented)
+      return
+    }
```

Comment:
Happily, in Swift you don't need `()` around conditionals in things like if, for, while, etc. And, this should really be a switch:
```
switch call.method {
case "startSensorReadings": 
  startSensorReadings()
  result("success")
...
default:
  result(FlutterMethodNotImplemented)
  // The return in your code is not needed)
}
```